src/strpkg/extract.nim

```diff
@@ -15,42 +16,21 @@ import strformat
 import math
 import argparse
 
-# This is the bottleneck for run time at the moment
-proc get_repeat(read: var string, counts: var Seqs[uint8], repeat_count: var int, opts:Options): array[6, char] =
-  repeat_count = 0
-  if read.count('N') > 20: return
-  var s = newString(6)
-
-  var best_score: int = -1
-  for k in 2..6:
-    let count = read.count(k, counts[k])
-    var score = count * k
-    if score <= best_score:
-      if count < (read.len.float * 0.12 / k.float).int:
-        break
-      continue
-    best_score = score
-    if count > (read.len.float * opts.proportion_repeat / k.float).int:
-      # now check the actual string because the kmer method can't track phase
-      s = newString(k)
-      counts[k].argmax.decode(s)
-      if read.count(s) > (read.len.float * opts.proportion_repeat / k.float).int:
-        copyMem(result.addr, s[0].addr, k)
-        repeat_count = count
-    elif count < (read.len.float * 0.12 / k.float).int:
-      # e.g. for a 5 mer repeat, we should see some 2, 3, 4 mer reps and we can
-      # bail if we do not. this is an optimization to avoid counting when we
-      # cant possibly see a repeat.
-      break
-
-proc get_repeat*(aln:Record, counts:var Seqs[uint8], repeat_count: var int, align_length: var int, opts:Options): array[6, char] =
+proc get_repeat*(aln:Record, genome_str:TableRef[string, Lapper[region]], counts:var Seqs[uint8], repeat_count: var int, align_length: var int, opts:Options): array[6, char] =
   when defined(skipFullMatch):
-    # TODO: write the code to make find repetitive regions in fasta and use
-    # that with this as default.
     # compile with -d:skipFullMatch to make it much faster
```

Comment:
Is there an advantage to having this option set at compile rather than as a command line option?

Reply:
this could be removed as the new genome (fasta) scan works almost as fast as this.